src/plugins/discover/public/application/main/services/discover_state.ts

```diff
@@ -198,6 +204,7 @@ export function getDiscoverStateContainer({
   const stateStorage = createKbnUrlStateStorage({
     useHash: storeInSessionStorage,
     history,
+    useHashQuery: mode !== 'embedded',
```

Comment:
Not necessarily for this PR, but you may want to consider how this relates to the Discover locators that use `setStateToKbnUrl` which also relies on a `useHashQuery` param to construct the URLs.

Reply:
Good point. Thanks. Will take a look.